pkg/reconciler/buildrun/resources/taskrun.go

```diff
@@ -215,7 +217,31 @@ func GenerateTaskSpec(
 		amendTaskSpecWithImageMutate(cfg, &generatedTaskSpec, build.Spec.Output)
 	}
 
-	return &generatedTaskSpec, nil
+	// Combine the environment variables specified in the Build object and the BuildRun object
+	// env vars in the BuildRun supercede those in the Build, overwriting them
+	combinedEnvs, err := env.MergeEnvVars(buildRun.Spec.Env, build.Spec.Env, true)
+	if err != nil {
+		return nil, err
+	}
+
+	// errs holds a slice of error objects from the merge process
+	var errs []error
+
+	// Iterate over the Steps of the BuildStrategy and merge in the combined environment
+	// variables from the Build/BuildRun
+	for i, step := range generatedTaskSpec.Steps {
```

Comment:
At this point in time, the Task spec contains not just the steps from the build strategy but also those from the sources and the mutate image step depending on the Build. You'll have to move this up into the loop after the `// define the steps coming from the build strategy` comment to only modify build strategy steps.

Reply:
updated in next push

Reply:
Updated in the current push, ptal.